src/collective/excelexport/datasources/base.py

```diff
@@ -71,7 +71,7 @@ def sort_exportables(self, exportables):
             exportables_dict = OrderedDict()
             for x in exportables:
                 # we use list because there can be several exportables with same name
-                exportables_dict.setdefault(get_name(x), []).append(x)
+                exportables_dict.setdefault(getattr(x, 'name', None) or get_name(x), []).append(x)
```

Comment:
Why don't you have this x.name check in get_name function?
Or maybe get_name is used elsewhere and you don't want the check in get_name.